### DISCUSSION TITLE
Identify Signature M1 x Intel

### DISCUSSION THREAD

**Discussion by daniloak** [ImageMagick/ImageMagick#5720] [Help]
Does the signature change if I am using a M1 or an Intel chip?
I am running `identify ae884bb8-ee63-4bc0-b93b-1565c5053294.jpg -verbose` and it seems the signature is different depending on the computer, I might be wrong in this assumption
I get f7b8a72449c3000d8e3bfab41052fca715ff056dc6eeca0e3a4eaf1595241806

ImageMagick 6

![200031140-059fc8ce-5bd1-4ac3-983d-96c95e0309b8](https://user-images.githubusercontent.com/25535327/200041596-96f2845a-752e-4beb-8fbe-d08e5bc10964.jpg)

**Comment by fmw42** [comment]
Pardon me, but signature of what? This question seems to be more of a Mac question. Please clarify further.

**Reply by daniloak** [reply]
I updated the question, sorry

**Comment by urban-warrior** [comment]
You will get the same signature for any image with no or lossless compression.  JPEG has lossy compression and may return different results depending on the JPEG library and different factors in its build and may return different signatures as a result on different hosts.

**Comment by wkdhkr** [comment]
Why is it difficult to get the same signature universally across CPU architectures?
Even in the same environment, different versions of ImageMagick may return different signatures.
I was looking for a signature that can be verified independently of the environment, like file md5 and sha1, but ImageMagick's signature doesn't seem to be suitable. Are there alternatives?

**Reply by wkdhkr** [reply]
Related topics:
https://github.com/ImageMagick/ImageMagick/discussions/2970

**Comment by urban-warrior** [comment]
ImageMagick v7 should return the same signature across CPU architectures if it follows the IEEE floating point standard.  Can you provide an image that produces different image signatures for the latest release of ImageMagick?
